src/indexed_crate.rs

```diff
@@ -231,154 +231,139 @@ pub struct IndexedCrate<'a> {
+                        acc.proc_macros.insert(item.id, item);
                     }
                     _ => {}
                 }
             }
 
             acc
-        });
-
-        #[cfg(feature = "rayon")]
-        let folded = {
-            let collected = folded
-                .map(|index| vec![index])
-                .reduce(Vec::new, |mut left, right| {
-                    left.extend(right);
-                    left
-                });
-
-            let mut fns_len = 0;
-            let mut structs_len = 0;
-            let mut enums_len = 0;
-            let mut unions_len = 0;
-            let mut traits_len = 0;
-            let mut modules_len = 0;
-            let mut statics_len = 0;
-            let mut consts_len = 0;
-            let mut decl_macros_len = 0;
-            let mut proc_macros_len = 0;
-
-            for c in &collected {
-                fns_len += c.free_functions.len();
-                structs_len += c.structs.len();
-                enums_len += c.enums.len();
-                unions_len += c.unions.len();
-                traits_len += c.traits.len();
-                modules_len += c.modules.len();
-                statics_len += c.statics.len();
-                consts_len += c.free_consts.len();
-                decl_macros_len += c.decl_macros.len();
-                proc_macros_len += c.proc_macros.len();
-            }
-
-            let mut value = Self {
-                free_functions: Vec::with_capacity(fns_len),
-                structs: Vec::with_capacity(structs_len),
-                enums: Vec::with_capacity(enums_len),
-                unions: Vec::with_capacity(unions_len),
-                traits: Vec::with_capacity(traits_len),
-                modules: Vec::with_capacity(modules_len),
-                statics: Vec::with_capacity(statics_len),
-                free_consts: Vec::with_capacity(consts_len),
-                decl_macros: Vec::with_capacity(decl_macros_len),
-                proc_macros: Vec::with_capacity(proc_macros_len),
-            };
+        })
+    }
 
-            for c in collected {
-                value.free_functions.extend(c.free_functions);
-                value.structs.extend(c.structs);
-                value.enums.extend(c.enums);
-                value.unions.extend(c.unions);
-                value.traits.extend(c.traits);
-                value.modules.extend(c.modules);
-                value.statics.extend(c.statics);
-                value.free_consts.extend(c.free_consts);
-                value.decl_macros.extend(c.decl_macros);
-                value.proc_macros.extend(c.proc_macros);
+    /// Returns true if the index corresponding to the type given by
+    /// `destination_type` contains an element with the id `item_id`.
+    /// Conservatively, returns true if the destination is None or of a type
+    /// that isn't contained in the indexes.
+    pub fn contains(&self, destination_type: Option<Arc<str>>, item_id: Id) -> bool {
```

Comment:
We don't use the fact that we get an `Arc` so let's not demand it here. Otherwise we're tying up the caller's ability to change implementations needlessly.
```suggestion
    pub fn contains(&self, destination_type: Option<&str>, item_id: Id) -> bool {
```